src/Payments/Integrations/CashOnDelivery.php

```diff
@@ -0,0 +1,103 @@
+<?php
+/**
+ * Cash on Delivery (COD) (core) gateway implementation.
+ *
+ * @package WooCommerce/Blocks
+ * @since 2.10.0
+ */
+
+namespace Automattic\WooCommerce\Blocks\Payments\Integrations;
+
+use Automattic\WooCommerce\Blocks\Assets\Api;
+
+/**
+ * Cash on Delivery (COD) payment method integration
+ *
+ * @since 2.10.0
+ */
+final class CashOnDelivery extends AbstractPaymentMethodType {
+	/**
+	 * Payment method name/id/slug (matches id in WC_Gateway_COD in core).
+	 *
+	 * @var string
+	 */
+	protected $name = 'cod';
+
+	/**
+	 * An instance of the Asset Api
+	 *
+	 * @var Api
+	 */
+	private $asset_api;
+
+	/**
+	 * Constructor
+	 *
+	 * @param Api $asset_api An instance of Api.
+	 */
+	public function __construct( Api $asset_api ) {
+		$this->asset_api = $asset_api;
+	}
+
+	/**
+	 * Initializes the payment method type.
+	 */
+	public function initialize() {
+		$this->settings = get_option( 'woocommerce_cod_settings', [] );
+	}
+
+	/**
+	 * Returns if this payment method should be active. If false, the scripts will not be enqueued.
+	 *
+	 * @return boolean
+	 */
+	public function is_active() {
+		return ! empty( $this->settings['enabled'] ) && 'yes' === $this->settings['enabled'];
```

Comment:
`filter_var( $this->settings['enabled'], FILTER_VALIDATE_BOOLEAN )`  (here and elsewhere) is a nice way to handle this kind of thing as it explicitly allows for various boolean type indicators ('yes', 'no', etc). It'd also be future proof if Woo ever changed the settings values to actual booleans (or 1/0).

Reply:
Good point - fixed in all the core payment methods :) e504ba352568eecaebd1705842cbce63aa05fcb3